sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py

```diff
@@ -214,16 +214,20 @@ async def receive_messages_async(consumer, batch, max_batch_size, max_wait_time)
         # pylint:disable=protected-access
         consumer._callback_task_run = True
         consumer._last_callback_called_time = time.time()
-        callback_task = asyncio.ensure_future(
+        callback_task = asyncio.create_task(
             PyamqpTransportAsync._callback_task(consumer, batch, max_batch_size, max_wait_time)
         )
-        receive_task = asyncio.ensure_future(PyamqpTransportAsync._receive_task(consumer))
+        receive_task = asyncio.create_task(PyamqpTransportAsync._receive_task(consumer))
 
-        try:
-            await receive_task
-        finally:
-            consumer._callback_task_run = False
```

Comment:
What was this check for?

Reply:
I think its used to break the callback [task](https://github.com/Azure/azure-sdk-for-python/blob/5013c2d8464a518ff4f2a6ca691f7913e1620c6e/sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py#L151) out from the infinite while loop so it could be awaited.